packages/ui/src/models/visualization/flows/kamelet-visual-entity.ts

```diff
@@ -28,17 +31,29 @@ export class KameletVisualEntity extends AbstractCamelVisualEntity {
 
   getComponentSchema(path?: string | undefined): VisualComponentSchema | undefined {
     if (path === ROOT_PATH) {
-      /** A better schema will be provided at a later stage */
       return {
         title: 'Kamelet',
-        schema: {} as KaotoSchemaDefinition['schema'],
+        schema: this.getRootKameletSchema(),
         definition: this.route,
       };
     }
 
     return super.getComponentSchema(path);
   }
 
+  private getRootKameletSchema(): KaotoSchemaDefinition['schema'] {
+    const processorDefinition = CamelCatalogService.getComponent(CatalogKind.Entity, 'KameletConfiguration');
+
+    if (processorDefinition === undefined) return {} as unknown as KaotoSchemaDefinition['schema'];
+
+    let schema = {} as unknown as KaotoSchemaDefinition['schema'];
+    if (processorDefinition.propertiesSchema !== undefined) {
+      schema = cloneDeep(processorDefinition.propertiesSchema);
+    }
```

Comment:
In this case, we don't need to clone the `schema` because we're not altering it like we're doing with a Processor + Component schema.

Reply:
Ah, This is me just copy-pasting.